medmodels/matching/metrics.py

```diff
@@ -63,6 +65,43 @@ def exact_metric(vector1: NDArray[np.float64], vector2: NDArray[np.float64]) ->
     return np.inf
 
 
+def mahalanobis_metric(*vectors: NDArray[np.float64], inv_cov: NDArray[np.float64]):
+    """
+    Returns mahalanobis metric for matching. Works better with continuous covariates.
+
+    .. math::
+        D(x, y) = \\sqrt{(x-y)^T S^{-1} (x-y)} where S is the covariate matrix of
+        the whole distribution
```

Comment:
Does this math equation work (in the context of formatting)? I've never seen it written like this but IDK

Reply:
[Wikipedia article on Mahalanobis distance](https://en.wikipedia.org/wiki/Mahalanobis_distance)
![image](https://github.com/limebit/medmodels/assets/60970451/9f874505-55e0-4a1f-a3af-0bc89540d72e)

And According to ChatGPT:
![image](https://github.com/limebit/medmodels/assets/60970451/c1fb31d0-e35c-4c38-940f-a96f6a0d38a7)